### PR TITLE
feat(room): `set_avatar_url` `remove_avatar_url` `upload_avatar` `set_topic`

### PR DESCRIPTION
**Pull request by Velin92**
Created the functions in the title, and exposed a binding for `upload_avatar` and `set_topic` in FFI

### CODE REVIEW

File: crates/matrix-sdk/src/room/joined.rs [hunk 1]
```diff
@@ -858,6 +863,61 @@ impl Joined {
         self.send_state_event(RoomPowerLevelsEventContent::from(power_levels)).await
     }
 
+    /// Sets the new topic for this room.
```
Comment (jplatte) [inline crates/matrix-sdk/src/room/joined.rs:866]:
```suggestion
    /// Sets a new topic for this room.
```

File: bindings/matrix-sdk-ffi/src/room.rs [hunk 2]
```diff
@@ -20,11 +20,14 @@ use matrix_sdk::{
         EventId, UserId,
     },
 };
+use mime::Mime;
 use tracing::error;
 
 use super::RUNTIME;
 use crate::{TimelineDiff, TimelineItem, TimelineListener};
 
+pub struct ImageInfo {}
```
[the remark this answers is not in the document]
Comment (Velin92) [reply bindings/matrix-sdk-ffi/src/room.rs:29]: oh yeah, let me remove this

[the remark this answers is not in the document]
Comment (jplatte) [reply bindings/matrix-sdk-ffi/src/room.rs:29]: I don't really care personally. So we can defer it, although the place where we pass `None` should maybe have a todo / fixme comment so it's clear we want to handle it differently in the future.

[the remark this answers is not in the document]
Comment (Velin92) [reply bindings/matrix-sdk-ffi/src/room.rs:29]: perfect! then removing this for now and adding a comment

**Review by jplatte — APPROVED** [review]

**Pull request merged**